spring-vault-core/src/main/java/org/springframework/vault/support/CertificateBundle.java

```diff
@@ -214,6 +227,41 @@ public KeyStore createKeyStore(String keyAlias, boolean includeCaChain) {
 		}
 	}
 
+	/**
+	 * Create a {@link KeyStore} from this {@link CertificateBundle} containing the
+	 * private key and certificate chain.
+	 * @param keyAlias the key alias to use.
+	 * @param includeCaChain whether to include the certificate authority chain instead of
+	 * just the issuer certificate.
+	 * @param password the password to use.
+	 * @return the {@link KeyStore} containing the private key and certificate chain.
+	 * @since 3.0.0
+	 */
+	public KeyStore createKeyStore(String keyAlias, boolean includeCaChain, String password) {
```

Comment:
It makes sense to accept `CharSequence` instead of `String` as password (and have another overload accepting `char[]`) as `String`s are heavily cached by the JVM. Also, the new method looks 90% similar to the already existing one. There's room to eliminate duplicate code by having an implementation that can handle absence/presence of the password.

Reply:
changes are committed and ready for review.